src/harness/virtualFileSystemWithWatch.ts

```diff
@@ -279,170 +279,42 @@ interface Array<T> { length: number; [n: number]: T; }`
-    }
-
-    export interface TestFsWatcher {
-        cb: FsWatchCallback;
-        directoryName: string;
-        fallbackPollingInterval: PollingInterval;
-        fallbackOptions: WatchOptions | undefined;
-    }
-
-    export interface ReloadWatchInvokeOptions {
-        /** Invokes the directory watcher for the parent instead of the file changed */
-        invokeDirectoryWatcherInsteadOfFileChanged: boolean;
-        /** When new file is created, do not invoke watches for it */
-        ignoreWatchInvokedWithTriggerAsFileCreate: boolean;
-        /** Invoke the file delete, followed by create instead of file changed */
-        invokeFileDeleteCreateAsPartInsteadOfChange: boolean;
-    }
-
-    export enum Tsc_WatchFile {
-        DynamicPolling = "DynamicPriorityPolling",
-        SingleFileWatcherPerName = "SingleFileWatcherPerName"
-    }
-
-    export enum Tsc_WatchDirectory {
-        WatchFile = "RecursiveDirectoryUsingFsWatchFile",
-        NonRecursiveWatchDirectory = "RecursiveDirectoryUsingNonRecursiveWatchDirectory",
-        DynamicPolling = "RecursiveDirectoryUsingDynamicPriorityPolling"
-    }
-
     export const timeIncrements = 1000;
-    export interface TestServerHostOptions {
-        useCaseSensitiveFileNames: boolean;
-        executingFilePath: string;
-        currentDirectory: string;
-        fileOrFolderorSymLinkList: readonly FileOrFolderOrSymLink[];
-        newLine?: string;
-        useWindowsStylePaths?: boolean;
-        environmentVariables?: ESMap<string, string>;
-    }
-
-    export class TestServerHost implements server.ServerHost, FormatDiagnosticsHost, ModuleResolutionHost {
-        args: string[] = [];
 
+    export class TestServerHost extends VirtualServerBaseHost implements server.ServerHost {
         private readonly output: string[] = [];
-
-        private fs: ESMap<Path, FSEntry> = new Map();
         private time = timeIncrements;
-        getCanonicalFileName: (s: string) => string;
-        private toPath: (f: string) => Path;
         private timeoutCallbacks = new Callbacks(this);
         private immediateCallbacks = new Callbacks(this);
         readonly screenClears: number[] = [];
-
-        readonly watchedFiles = createMultiMap<Path, TestFileWatcher>();
-        readonly fsWatches = createMultiMap<Path, TestFsWatcher>();
-        readonly fsWatchesRecursive = createMultiMap<Path, TestFsWatcher>();
-        runWithFallbackPolling: boolean;
-        public readonly useCaseSensitiveFileNames: boolean;
-        public readonly newLine: string;
-        public readonly windowsStyleRoot?: string;
         private readonly environmentVariables?: ESMap<string, string>;
-        private readonly executingFilePath: string;
-        private readonly currentDirectory: string;
         public require: ((initialPath: string, moduleName: string) => RequireResult) | undefined;
+        private readonly tscWatchFile?: string;
+        private readonly tscWatchDirectory?: string;

```

Comment:
Why do you need these two as private members when they are needed only for the construction of watchFile and watchDIrectory ?

Reply:
It was leftover from my previous class design; I thought I still needed it, but it turns out I don't.